File: giao_huu_bong_da/giao_huu_bong_da.cpp
#include<bits/stdc++.h>
using namespace std;
int N, x, y;
int res[101];
int cnt;

bool check(int v, int k, int l){
    if (k == x && l == y) return false;
    return true;
}

void solution(){
    cnt++;
}

void Try(int k, int l){
    for (int v = 0; v <= 1; v++){
        if (check(v, k, l)){
            if (v == 0 && k < x){
                k++;
                if (k == x || l == y) solution();
                else Try(k, l);
                k--;
            }
            if (v == 1 && l < y){
                l++;
                if (k == x || l == y) solution();
                else Try(k, l);  
                l--;            
            }
        }
    }
}

int main(){
    cin >> N;
    for (int i = 0; i < N; i++){
        cin >> x >> y;
        cnt = 0;
        Try(0, 0);
        res[i] = cnt;
    }
    for (int i = 0; i < N; i++){
        cout << res[i] << endl;
    }
    return 0;
}

File: giao_huu_bong_da/giao_huu_bong_da.py
def check(v, k, l, x, y):
    if k == x and l == y:
        return False
    return True

def solution():
    global cnt
    cnt += 1

def Try(k, l, x, y):
    for v in range(2):
        if check(v, k, l, x, y):
            if v == 0 and k < x:
                k += 1
                if k == x or l == y:
                    solution()
                else:
                    Try(k, l, x, y)
                k -= 1
            if v == 1 and l < y:
                l += 1
                if k == x or l == y:
                    solution()
                else:
                    Try(k, l, x, y)
                l -= 1

if __name__ == "__main__":
    N = int(input())
    res = []
    for _ in range(N):
        x, y = map(int, input().split())
        if x == 0 and y == 0:
            res.append(1)
        else:
            cnt = 0
            Try(0, 0, x, y)
            res.append(cnt)

    for r in res:
        print(r)
